jujugui/static/gui/src/app/components/profile/credential-list/credential-list.js

```diff
@@ -115,6 +125,70 @@ class ProfileCredentialList extends React.Component {
     });
   }
 
+  /**
+    Show the add credentials form.
+  */
+  _toggleAdd() {
+    this.setState({showAdd: !this.state.showAdd});
+  }
+
+  /**
+    Generate a form to add credentials.
+  */
+  _generateAddCredentials() {
+    return (
+      <ExpandingRow
+        classes={{'twelve-col': true}}
+        clickable={false}
+        expanded={this.state.showAdd}>
+        <div></div>
+        <div className="twelve-col">
+          {this._generateDeploymentCredentialAdd()}
+        </div>
+      </ExpandingRow>);
+  }
+
+  /**
+    Handle a credential having been created.
+    @param credential {String} The name of the newly created credential.
+  */
+  _onCredentialAdded(credential) {
+    // Load the credentials again so that the list will contain the newly
+    // added credential.
+    this._getClouds();
+    this._toggleAdd();
+  }
+
+  /**
+    Generate the add credentials UI with any supplied overrides depending
+    on where it is to be rendered.
+    @param {Object} overrides The overrides for the default props.
+    @return {Object} React component for DeploymentCredentialAdd
+  */
+  _generateDeploymentCredentialAdd(overrides={}) {
+    const controllerAPI = this.props.controllerAPI;
+    const credentials = this.state.credentials;
+    return (
+      <CredentialAddEdit
+        key="deployment-credential-add"
+        acl={this.props.acl}
+        addNotification={this.props.addNotification}
+        controllerAPI={shapeup.addReshape({
+          listClouds: controllerAPI.listClouds.bind(controllerAPI),
+          updateCloudCredential: controllerAPI.updateCloudCredential.bind(controllerAPI)
+        })}
+        controllerIsReady={this.props.controllerIsReady}
```

Comment:
Is this because we're reusing the deployment flow components and those require the live controller?

Reply:
That's correct. The component for choosing the cloud waits for the controller to be ready before requesting the list of clouds.